### PR TITLE
Removed switching event processor. 

### LINKED ISSUE #45: Revisit the switching event processor
The switching event processor feels like we're pushing back against fundamental design choices encouraged by Axon. We should revisit it and consider only keeping the replay functionality.

What benefit do we get from using tracking event processors for replays given that they only work when assigned to the default group?

### PR DESCRIPTION
**Pull request by sluehr**
It wasn't really providing much benefit and it's good practice to default to the tracking event processor

Closes #45 

## Proposed changes

The switching event processor feels like we're pushing back against fundamental design choices encouraged by Axon. We should revisit it and consider only keeping the replay functionality.

I've tested the reworked saga example in the `feature/better-sagas` branch and found that a long running saga interrupted by application shut down or crash:
 - will not recover when we're using subscribing event processors
 - will automatically continue processing the saga when the application restarts when using the tracking event processor

Worse, still, the event being handled by the saga is lost when the subscribing one is being used.

Tracking event processing seems to be the way to go. The switching one might as well be retired.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] My local build passed (You ran `./gradlew clean build` without failures)
- [x]I have added tests that prove my fix is effective or that my feature works
- [x]  I have added necessary documentation (if appropriate)
- [x] I have assigned a label to the PR

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -254,12 +254,11 @@ The starter kit comes with programmatic support for triggering replays. To perfo
...
-   [tracking event processors](https://axoniq.io/blog-overview/tracking-event-processors)
- * clearing the tracking tokens in the Axon database
- * placing a marker event into the event log that will end replays
- * starting TEP processing, and
- * changing the event processing configuration back to subscribing mode when the TEP reaches the marker event
+ * shutting down [tracking event processors](https://axoniq.io/blog-overview/tracking-event-processors) (TEP) processing;
+ * clearing the tracking tokens in the Axon database;
+ * placing a marker event into the event log that will end replays;
+ * starting TEP processing; and
+ * changing the event processing configuration back to subscribing mode when the TEP reaches the marker event.
```
Comment (ywangd) [inline README.md]: I thought we don't want to use Subscribing processor anymore? Also isn't the word subscribing "mode" a feature of the switching processor? I guess this bullet point can be removed?

File: src/axon-support/src/main/java/engineering/everest/lhotse/axon/replay/ReplayEndpoint.java [hunk 2]
```diff
@@ -58,25 +56,26 @@ public ReplayEndpoint(AxonConfiguration axonConfiguration,
     @WriteOperation
     public void startReplay(@Nullable Set<String> processingGroups,
                             @Nullable OffsetDateTime startTime) {
-        var replayableEventProcessors = processingGroups == null
-                ? getReplayableEventProcessors() : getReplayableEventProcessor(processingGroups);
+        var resettableEventProcessors = processingGroups == null
```
Comment (ywangd) [inline src/axon-support/src/main/java/engineering/everest/lhotse/axon/replay/ReplayEndpoint.java]: I don't have strong view for the renaming from `replayable` to `resettable`. But if it is intentional, can we make the change consistently across all relevant classes, methods and error/logging messages?

File: src/axon-support/src/main/java/engineering/everest/lhotse/axon/replay/ReplayMarkerAwareTrackingEventProcessor.java [hunk 3]
```diff
@@ -187,25 +167,14 @@ public Builder trackingEventProcessorConfiguration(
             return this;
         }
 
-        @Override
-        public Builder storingTokensAfterProcessing() {
-            super.storingTokensAfterProcessing();
-            return this;
-        }
```
Comment (ywangd) [inline src/axon-support/src/main/java/engineering/everest/lhotse/axon/replay/ReplayMarkerAwareTrackingEventProcessor.java:194]: I don't remember why we had this before. I guess it's not important anymore

File: src/axon-support/src/main/java/engineering/everest/lhotse/axon/config/AxonConfig.java [hunk 4]
```diff
@@ -104,7 +104,6 @@ public AnnotationCommandTargetResolver annotationCommandTargetResolver() {
 
     public enum EventProcessorType {
         SUBSCRIBING,
```
Comment (ywangd) [inline src/axon-support/src/main/java/engineering/everest/lhotse/axon/config/AxonConfig.java:106]: Given the issues you identified for subscribing procoessors and the fact we don't expose a way to replay with it, should we also remove support for subscribing processor altogether? It can be a separate PR.

**Review by ywangd — APPROVED** [review]
Looks good to me.

File: README.md (hunk 1 above)
Comment (sluehr) [reply README.md]: :blush:

File: src/axon-support/src/main/java/engineering/everest/lhotse/axon/replay/ReplayEndpoint.java (hunk 2 above)
Comment (sluehr) [reply src/axon-support/src/main/java/engineering/everest/lhotse/axon/replay/ReplayEndpoint.java]: There was a reason but then I backed out. This is a vestigial change.

File: src/axon-support/src/main/java/engineering/everest/lhotse/axon/config/AxonConfig.java (hunk 4 above)
Comment (sluehr) [reply src/axon-support/src/main/java/engineering/everest/lhotse/axon/config/AxonConfig.java:106]: It's available in Axon.... I don't recommend using it going forward.... maybe? :-)
